source/locationHelper.py

```diff
@@ -0,0 +1,398 @@
+
+	def __ge__(self,other):
+		"""
+		Returns whether self is greater than or equal to other.
+		This first compares y, then x coordinates.
+		For example: (x=3,y=4) > (x=4,y=3) because self.y is greater than other.y.
+		To compare in opposite order (i.e. compare x, then y), use tuple(self) >= tuple(other)
+		"""
+		if not isinstance(other,POINT_CLASSES):
+			try:
+				other=toPoint(other)
+			except ValueError:
+				return False
+		return (self.y, self.x) >= (other.y, other.x)
+
+	def __eq__(self,other):
+		if not isinstance(other,POINT_CLASSES):
+			try:
+				other=toPoint(other)
+			except ValueError:
+				return False
+		return self.x==other.x and self.y==other.y
+
+	def __neq__(self,other):
+		return not (self==other)
+
+	def toPOINT(self):
+		"""Converts self to a L{ctypes.wintypes.POINT}"""
+		return POINT(self.x,self.y)
+
+	def toLogical(self, hwnd):
+		"""Converts self from physical to logical coordinates and returns a new L{Point}."""
+		return Point(*windowUtils.physicalToLogicalPoint(hwnd, *self))
+
+	def toPhysical(self, hwnd):
+		"""Converts self from logical to physical coordinates and returns a new L{Point}"""
+		return Point(*windowUtils.logicalToPhysicalPoint(hwnd, *self))
+
+	def toClient(self, hwnd):
+		"""Converts self from screen to client coordinates and returns a new L{Point}"""
+		return Point(*winUser.ScreenToClient(hwnd, *self))
+
+	def toScreen(self, hwnd):
+		"""Converts self from client to screen coordinates and returns a new L{Point}"""
+		return Point(*winUser.ClientToScreen(hwnd, *self))
+
+class _RectMixin:
+	"""Mix-in class for properties shared between Location and Rect classes"""
+
+	def toRECT(self):
+		"""Converts self to a L{ctypes.wintypes.RECT}"""
+		return RECT(self.left,self.top,self.right,self.bottom)
+
+	def toLogical(self, hwnd):
+		left,top=self.topLeft.toLogical(hwnd)
+		right,bottom=self.bottomRight.toLogical(hwnd)
+		if isinstance(self, Location):
+			return Location(left,top,right-left,bottom-top)
+		return Rect(left,top,right,bottom)
+
+	def toPhysical(self, hwnd):
+		left,top=self.topLeft.toPhysical(hwnd)
+		right,bottom=self.bottomRight.toPhysical(hwnd)
+		if isinstance(self, Location):
+			return Location(left,top,right-left,bottom-top)
+		return Rect(left,top,right,bottom)
+
+	def toClient(self, hwnd):
+		left, top =self.topLeft.toClient(hwnd)
+		if isinstance(self, Location):
+			return Location(left, top, self.width, self.height)
+		return Rect(left, top, left+self.width, top+self.height)
+
+	def toScreen(self, hwnd):
+		left,top=self.topLeft.toScreen(hwnd)
+		if isinstance(self, Location):
+			return Location(left, top, self.width, self.height)
+		return Rect(left, top, left+self.width, top+self.height)
+
+	@property
+	def topLeft(self):
+		return Point(self.left,self.top)
+
+	@property
+	def bottomRight(self):
+		return Point(self.right,self.bottom)
+
+	@property
+	def center(self):
+		return Point((self.left+self.right)/2, (self.top+self.bottom)/2)

```

Comment:
Can you add a test for this where `self.left = -5`, `self.right=5`, `self.top=5`, `self,bottom=-5`

Reply:
I should have been more specific here. I was worried about 0 divided by 2. You can ignore this.

Thinking about this more carefully, there are other concerns:
- A minor one, is performance. Often multiplication is faster than division, though we are unlikely to be doing this so many times an update that this would be at all noticeable.
- Rounding, given `RectLTRB(0,2,2,0).center()` gives `Point(1,1)` and `RectLTRB(0,3,3,0).center()` also gives `Point(1, 1)` callers may be expecting floating point numbers to be returned.
- A bug: `RectLTRB(10,13,13,10).center()` gives `Point(1,1)` when it should give `Point(11,11)`
  Change the implementation of `center` to `Point( self.left+self.width/2, self.top-self.height/2)`
  Please add a test for this, and also for the same in the negative quadrant